core/dbt/parser/schemas.py

```diff
@@ -325,8 +325,6 @@ def _parse_generic_test(
             'kwargs': builder.args,
         }
         tags = sorted(set(itertools.chain(tags, builder.tags())))
-        if 'schema' not in tags:
```

Comment:
This seems randomly removed but maybe it isn't?

Reply:
Super fair question!

In v0.18, we introduced a selection method, `test_type:`, that allowed users to select just schema tests (`dbt test --models test_type:schema`) or data tests (`dbt test --models test_type:data`).

Prior to that, we supported the same functionality, but in a pretty silly way. dbt would secretly add tags—`'schema'` to schema tests, `'data'` to data tests—and then users could pass a flag `dbt test --schema` to select all tests tagged "schema," or `dbt test --data` to select all tests tagged "data." Funny enough, that would include schema tests manually tagged "data," too, or data tests that selected from models tagged "schema." Not a good system.

This PR takes the opinionated view that we should do away with that legacy approach to the functionality, since we've had a better approach (`test_type` selector) for several versions now. To that end, there's some backwards compatibility for the newer approach, and total deprecation fro the older approach:
- Stop secretly adding `'schema'` and `'data'` tags to tests
- Deprecate `--schema` and `--data` flags entirely
- Continue supporting `test_type:schema` + `test_type:data` as aliases for `test_type:generic` and `test_type:singular`, respectively